### DISCUSSION TITLE
Setting up with jest & testing-library/react?

### DISCUSSION THREAD

**Discussion by tbrasington** [system-ui/theme-ui#1927] [Q&A] [answered]
Hello

I have run into a challenge with getting testing to work with theme-ui. 

On running the tests I get this error
` TypeError: (0 , _jsxRuntime.jsx) is not a function`

Scratching my head here to what could be causing the issue.

There is a demo library here to see the error https://github.com/tbrasington/theme-test 

Any advice is greatly appreciated on how to do this.

T

**Comment by tbrasington** [comment, marked as answer]
So I got to the bottom of the issue:
As I was using aliases' I had to ensure that my babel, tsconfig and jest where setup to look for them.
I did abandon their use in the end as it was becoming a hassle to manage all the paths. If anyone finds a neat way of handling it, I am all ears.

**Reply by hasparus** [reply]
Hey 👋. I'm glad you managed to figure this out.

Just out of curiosity — did you try [TSConfig Paths](https://www.typescriptlang.org/tsconfig#paths)? 

I'm using them in client's project and the setup works pretty fine with [`ts-jest`](https://huafu.github.io/ts-jest/user/config/#jest-config-with-helper) and [`tsconfig-paths-webpack-plugin`](https://www.npmjs.com/package/tsconfig-paths-webpack-plugin).

**Reply by tbrasington** [reply]
I had, but I think the combination of all 3 is causing an issue. Need to get some spare time to dig into it as if I ran it with webpack it was all fine. Just wondering if that can be skipped.
